src/components/App.js

```diff
@@ -1,30 +1,57 @@
-import { TextField, Fab, Grid, InputLabel, OutlinedInput, InputAdornment, IconButton, VisibilityOff, FormControl, FormLabel, FormControlLabel, Button, Checkbox, Typography, Stepper, Step, StepLabel } from '@mui/material'
+import { TextField, Fab, Grid, InputLabel, OutlinedInput, InputAdornment, FormControl, FormControlLabel, Button, Checkbox, Typography, Stepper, Step, StepLabel } from '@mui/material'
 import AddIcon from '@mui/icons-material/Add'
-import Visibility from '@mui/icons-material/Visibility'
 import React, { useState } from "react"
+import { useAppData } from './context/hooks'
+import { MUTATION_KEYS, useMutation } from '../config/queryClient'
 
 function App() {
+  const { data } = useAppData()
+  const { mutate: postAppData } = useMutation(MUTATION_KEYS.POST_APP_DATA)
+
   const [question, setQuestion] = useState("");
-  const [choice1, setChoice1] = useState("");
-  const [choice2, setChoice2] = useState("");
+  const [choices, setChoices] = useState(["", ""]);
 
-  var choiceList = []
-  choiceList.push([choice1, setChoice1])
-  choiceList.push([choice2, setChoice2])
+  const [answers, setAnswers] = React.useState([{
+    answer: false
+  },
+  {
+    answer: false
+  }]);
 
-  //const [answers, setAnswers] = useState([{ value: 'it is answer 1', checked: false }])
+  const handleAnswerCorrectnessChange = (index, e) => {
+    let newAnswers = [...answers]
+    newAnswers[index].answer = e.target.checked
+    setAnswers(newAnswers)
+  };
 
-  const [answers, setAnswers] = React.useState({
-    answer1: false,
-    answer2: false,
-  });
+  const handleChoiceChange = (index, e) => {
+    let newChoices = [...choices]
+    newChoices[index] = e.target.value
+    setChoices(newChoices)
+  }
 
-  const { answer1, answer2 } = answers;
+  const addAnswer = () => {
+    setChoices([...choices, ""])
+    setAnswers([...answers, { answer: false }])
+  }
 
-  const handleAnswerCorrectnessChange = (event) => {
-    setAnswers({ ...answers, [event.target.name]: event.target.checked });
-  };
+  function getCorrectAnswersIndices(answers) {
+    let correctAnswers = []
+    answers.forEach((answer, index) => { if (answer.answer == true) correctAnswers.push(index) })
+    return correctAnswers
+  }
 
+  const onSave = () => {
+    postAppData({
+      id: data?.get(0).id,
+      data: {
+        question: question,
+        answers: choices,
```

Comment:
Let's not have both `answers` and `choices` as name, choose one and stick to it.

Reply:
Actually I see you're not very clear with the nomenclature. I'll suggest keeping `choices` and use `isCorrect` to define whether it's a valid choice. 
Instead of having two arrays for the string and the correctness, let's have one with the following structure:
```js
const [choices] = useState([{
  text: "My choice text",
  isCorrect: true/false
}, { --- }........])
```